cudax/include/cuda/experimental/__memory_resource/any_resource.cuh

```diff
@@ -235,6 +235,16 @@ public:
     *this      = _CUDA_VSTD::move(__tmp);
   }
 
+  //! @brief Equality comparison between two \c basic_any_resource
+  //! @param __rhs The other \c basic_any_resource
+  //! @return Checks whether both resources have the same equality function stored in their vtable and if so returns
+  //! the result of that equality comparison. Otherwise returns false.
+  _CCCL_NODISCARD bool operator==(const basic_any_resource& __rhs) const
+  {
+    return (this->__static_vtable->__equal_fn == __rhs.__static_vtable->__equal_fn)
+        && this->__static_vtable->__equal_fn(this->_Get_object(), __rhs._Get_object());
+  }
+
```

Comment:
why is this extra overload needed?

Reply:
I believe that was a fallout from the rework of how we determine whether two `any_resource`s are compatible.

In the tests I could not compare two identical `any_resource`s.

I thought that should have been covered here but 🤷

Reply:
i'd rather we got to the bottom of the problem.

Reply:
I agree, if it's not working for identical types, chances are it won't work for some other combinations and the overload is not enough here.

Reply:
I looked into it and this is needed to compare against an arbitrary resource

In that case we can utilize the conversion sequence. Previously it would fail, because we cannot deduce the template arguments there